--- final/Makefile
+++ temporary-coverage/Makefile
@@ -19,7 +19,7 @@
 CC := clang
 
 # These flags will be applied to your code any time it is built.
-CFLAGS := -Wall -std=c99 -D_POSIX_C_SOURCE=200809L
+CFLAGS := -Wall -std=c99 -D_POSIX_C_SOURCE=200809L -fprofile-arcs -ftest-coverage
 
 # These flags are applied only if you build your code with "make DEBUG=1".  -g
 # generates debugging symbols, -DDEBUG defines the preprocessor symbol "DEBUG"
@@ -40,7 +40,7 @@
 # If you need to link against libraries, add the appropriate flags here.  By
 # default, your code is linked against the "rt" library with the flag -lrt;
 # this library is used by the timing code in the testbed.
-LDFLAGS := -lrt -flto -fuse-ld=gold
+LDFLAGS := -lrt -flto -fuse-ld=gold -fprofile-arcs -ftest-coverage
 
 ################################################################################
 # You probably won't need to change anything below this line, but if you're
